components/KickoffRequestView/KickoffRequestPanel.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import { useRecoilValue } from 'recoil';
+import { selectedMeasureState } from '../../atoms/selectedMeasure';
+import { Card, Tabs, Title, Space } from '@mantine/core';
+import KickoffHeaders from './KickoffHeaders';
+import KickoffBody from './KickoffBody';
+import { exportUrlState } from '../../atoms/exportUrl';
+import KickoffPostUrl from './KickoffPostUrl';
+
+export default function KickoffRequestPanel() {
+  const selectedMeasure = useRecoilValue(selectedMeasureState);
+  const exportUrl = useRecoilValue(exportUrlState);
+  const [activeTab, setActiveTab] = useState(0);
+  let body: fhir4.Parameters | undefined;
+  let headers: string | undefined;
+
+  if (selectedMeasure && exportUrl) {
+    [body, headers] = buildKickoffRequest(selectedMeasure, exportUrl);
```

Comment:
While this works, I think it could be good to use a [recoil selector](https://recoiljs.org/docs/basic-tutorial/selectors/) for this, since it is a piece of state for this component that is derived from the values of other state.

This also helps us if we need to access this information in other components down the line (like a request history viewing or something, idk)

We can use the exact same function you have for building the kickoff request information, but just put it in a recoil selector and compute its value when selectedMeasure and exportUrl are defined. We can get their values using `get` like the recoil docs show.

Thoughts?

Reply:
Awesome suggestion. Implemented in bb84cdb.